cli/lib/rotate-key.js

```diff
@@ -0,0 +1,138 @@
+"use strict";
+
+const pem = require("pem");
+const util = require("util");
+const debug = require("debug")("jwkrotatekey");
+const commander = require('commander');
+const request = require("request");
+
+
+
+function createCert(cb) {
+
+    const options = {
+        selfSigned: true,
+        days: 1
+    };
+
+    pem.createCertificate(options, cb);
+}
+
+function generateCredentialsObject(options) {
+    if (options.token) {
+        return {
+            "bearer": options.token
+        };
+    } else {
+        return {
+            user: options.username,
+            pass: options.password
+        };
+    }
+}
+
+const RotateKey = function () {
+
+}
+
+module.exports = function () {
+  return new RotateKey();
+}
+
+RotateKey.prototype.rotatekey = function rotatekey(options, cb) {
+
+    options.baseuri = options.baseuri || "https://api.enterprise.apigee.com";
+    options.kvm = "microgateway";
+    options.kid =  options.kid || "2";
+
+    var privateKeyURI = util.format("%s/v1/organizations/%s/environments/%s/keyvaluemaps/%s/entries/private_key",
+        options.baseuri, options.org, options.env, options.kvm);
+    console.log("Checking if private key in the KVM...");
+    request({
+        uri: privateKeyURI,
+        auth: generateCredentialsObject(options),
+        method: "GET"
+    }, function(err, res, body) {
+        if (err) {
+            console.error(err);
+        } else {
+            console.log("Private key found");
+            var publicKeyURI = util.format("%s://%s-%s.apigee.net/edgemicro-auth/publicKey",
```

Comment:
The only issue that I see with this is that it will not work for on-prem customers.  It assumes apigee.net as the domain. I created a pull request to your rotate-key repo to handle on-prem customers.  Can we change this?